sys/net/gnrc/netif/ieee802154/gnrc_netif_ieee802154.c

```diff
@@ -265,14 +297,67 @@ static int _send(gnrc_netif_t *netif, gnrc_pktsnip_t *pkt)
         DEBUG("_send_ieee802154: Error preperaring frame\n");
         return -EINVAL;
     }
-
-    /* prepare iolist for netdev / mac layer */
-    iolist_t iolist = {
-        .iol_next = (iolist_t *)pkt->next,
+    mhr_len = res;
+#if IS_USED(MODULE_IEEE802154_SECURITY)
+    {
+        /* write protect `pkt` to set `pkt->next` */
+        gnrc_pktsnip_t *tmp = gnrc_pktbuf_start_write(pkt);
+        if (tmp == NULL) {
+            DEBUG("_send_ieee802154: no write access to pkt");
+            return -ENOMEM;
+        }
+        pkt = tmp;
+        tmp = gnrc_pktbuf_start_write(pkt->next);
+        if (tmp == NULL) {
+            DEBUG("_send_ieee802154: no write access to pkt->next");
+            return -ENOMEM;
+        }
+        pkt->next = tmp;
+        /* merge snippets to store the L2 payload uniformly in one buffer */
+        res = gnrc_pktbuf_merge(pkt->next);
+        if (res < 0) {
+            DEBUG("_send_ieee802154: failed to merge pktbuf\n");
+            return res;
+        }
+    }
+#endif
+    iolist_t iolist_payload = {
+        .iol_next = (iolist_t *)pkt->next->next,
+        .iol_base = pkt->next->data,
+        .iol_len = pkt->next->size
```

Comment:
Shouldn't this remain `pkt->next` for the non security case?

Reply:
If I change it to `pkt->next`, `ping6` no longer works. But I understand what you mean because in `master` there it is:
```
    iolist_t iolist = {
        .iol_next = (iolist_t *)pkt->next,
        .iol_base = mhr,
        .iol_len = (size_t)res
    };
```
To me it says: The L2 payload starts at `pkt->next`.
But this is the same as:
```
    iolist_t iolist_payload = {
        .iol_next = (iolist_t *)pkt->next->next,
        .iol_base = pkt->next->data,
        .iol_len = pkt->size
    };
```

Reply:
> If I change it to `pkt->next`, `ping6` no longer works. But I understand what you mean because in `master` there it is:

Ok. Just tested it myself. Confirmed! :-)